### PR TITLE
Run wasm-opt -O3 on CanvasKit wasm artifacts

### PR DESCRIPTION
**Pull request by hterkelsen**
Runs `wasm-opt -O3` on our CanvasKit wasm artifacts.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: third_party/canvaskit/run_wasm_opt.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  args = parser.parse_args()
+  try:
+    subprocess.check_call([
+        WASM_OPT_PATH,
+        '-O3',
+        args.input,
+        '-o',
+        args.output,
+    ],
+                          stdout=subprocess.DEVNULL)
```
Comment (kevmoo) [inline third_party/canvaskit/run_wasm_opt.py:34]: whitespace?
```suggestion
    stdout=subprocess.DEVNULL)
```

File: lib/web_ui/dev/steps/copy_artifacts_step.dart [hunk 2]
```diff
@@ -43,7 +43,7 @@ class CopyArtifactsStep implements PipelineStep {
     }
     if (artifactDeps.canvasKitChromium) {
       print('Copying CanvasKit (Chromium)...');
-      await copyCanvasKitFiles('canvaskit_chromium', 'canvaskit/chromium');
+      await copyCanvasKitFiles('canvaskit/chromium', 'canvaskit/chromium');
```
Comment (kevmoo) [inline lib/web_ui/dev/steps/copy_artifacts_step.dart:46]: why this change?

Comment (eyebrowsoffire) [reply lib/web_ui/dev/steps/copy_artifacts_step.dart:46]: This is a good change. My unit test reorganization has us pulling directly from the location of the original build output, which is changing as we mess around with the toolchain names and stuff. However, now we have copy steps in the build that put everything into the structure we'll see in GCS, so Harry is changing it so it pulls from that location. This is better and easier to maintain.

**Comment by kevmoo** [comment]
Are we still running with this @hterkelsen ?
